### PR TITLE
feat(browse-pagination): Added the browse pagination feature

### PR DESCRIPTION
**Pull request by Shruti3004**
## Description

Added the pagination feature in browse page.

### Changes

- Added the entries field.
- Added the pagination just like the current implementation.
- Stored the response header of x-total-pages in localStorage.

## How to test

 Visit to `/browse`

## Screenshots
Screenshot from 2021-07-06 03-14-53

### CODE REVIEW

**Comment by GMishx** [comment]
Unable to list pages even though `X-Total-Pages` says 10 pages are available.

image

**Comment by Shruti3004** [comment]
@GMishx please check it once, because for me it is working fine

File: src/services/browse.js [hunk 1]
```diff
@@ -17,10 +17,14 @@
...
 import { browseFiles } from "../api/browse";
+import { getLocalStorage } from "../shared/storageHelper";
 
 export function getBrowseData(browseData) {
   return browseFiles(browseData).then((res) => {
-    console.log(res);
-    return res;
+    const data = {
+      res,
+      pages: getLocalStorage("pages"),
```
Comment (GMishx) [inline src/services/browse.js:26]: I see, the code is reading pages from Local storage rather than the `X-Total-Pages` header?

Comment (GMishx) [reply src/services/browse.js:26]: My bad, it's stored here https://github.com/fossology/FOSSologyUI/pull/81/files#diff-e114ba983ca0ca43d47d5e491d386ff0e442b4483d8b8339160da54f57c4a852R61

**Comment by GMishx** [comment]
> @GMishx please check it once, because for me it is working fine, try logging it out.

Nope, still not working for me. Tried restarting the server, logout-login.

There is no "pages" in my local storage, even after reloading the page
image

**Comment by Shruti3004** [comment]
@GMishx I got the issue,

We need to set `Access-Control-Expose-Headers` in our backend to expose your custom response headers.
Taken from Mozilla Docs on `Access-Control-Expose-Headers:`

The Access-Control-Expose-Headers response header indicates which headers can be exposed as part of the response by listing their names.
By default, only the 6 CORS-safelisted response headers are exposed:
`Cache-Control`, `Content-Language`, `Content-Type`, `Expires`, `Last-Modified`, `Pragma`
If we want clients to be able to access other headers, you have to list them using the Access-Control-Expose-Headers header.

Therefore, we have to code your backend to set Access-Control-Expose-Headers such that it returns:
Access-Control-Expose-Headers: `x-total-pages`

**Comment by Shruti3004** [comment]
@GMishx Added the issue for [https://github.com/fossology/fossology/issues/2041(https://github.com/fossology/fossology/issues/2041) it and also opened the pr 
[https://github.com/fossology/fossology/pull/2042](https://github.com/fossology/fossology/pull/2042).

Till then,
you can test it with this chrome extension, [MOSEIF CORS](https://chrome.google.com/webstore/detail/moesif-origin-cors-change/digfbfaphojjndkpccljibejjbppifbc), just enable this extension and replace the value of `credentials` to `false` in `src/api/sendRequest.js`

**Pull request merged**